File: fix_trackplayer.js
const fs = require('fs');
const filepath = 'node_modules/react-native-track-player/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt';
const lines = fs.readFileSync(filepath, 'utf-8').split('\n');

let out = [];
let in_method = false;
let brace_depth = 0;

for (let line of lines) {
    if (line.includes(") = scope.launch {")) {
        line = line.replace(") = scope.launch {", ") { scope.launch {");
        in_method = true;
        brace_depth = (line.match(/\{/g) || []).length - (line.match(/\}/g) || []).length;
        out.push(line);
    } else if (in_method) {
        brace_depth += (line.match(/\{/g) || []).length - (line.match(/\}/g) || []).length;
        out.push(line);
        if (brace_depth === 1 && line.trim().startsWith("}")) {
            out.push("    }");
            in_method = false;
        }
    } else {
        out.push(line);
    }
}

fs.writeFileSync(filepath, out.join('\n'), 'utf-8');

File: fix_trackplayer.py
import sys

def fix_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        lines = f.readlines()
        
    out = []
    in_method = False
    brace_depth = 0
    
    for line in lines:
        if ") = scope.launch {" in line:
            line = line.replace(") = scope.launch {", ") { scope.launch {")
            in_method = True
            brace_depth = line.count('{') - line.count('}')
            out.append(line)
        elif in_method:
            brace_depth += line.count('{') - line.count('}')
            out.append(line)
            if brace_depth == 1 and line.strip().startswith("}"):
                out.append("    }\n")
                in_method = False
        else:
            out.append(line)
            
    with open(filepath, 'w', encoding='utf-8') as f:
        f.writelines(out)

fix_file('node_modules/react-native-track-player/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt')
